core/src/test/scala/com/phaller/rasync/test/PsSuite.scala

```diff
@@ -103,4 +103,29 @@ class PsSuite extends FunSuite {
     Await.ready(fut, 2.seconds)
   }
 
+  test("HandlerPool must be able to interrupt") {
+    implicit val pool = new HandlerPool(parallelism = 8)
+    val completer1 = CellCompleter[ReactivePropertyStoreKey, Int](new ReactivePropertyStoreKey())
+    val completer2 = CellCompleter[ReactivePropertyStoreKey, Int](new ReactivePropertyStoreKey())
+    val cell1 = completer1.cell
+    val cell2 = completer2.cell
+
+    cell2.whenNextSequential(cell1, v => {
+      NextOutcome(v)
+    })
+
+    pool.interrupt()
+    Thread.sleep(200)
+    completer1.putNext(10)
+
+    assert(cell2.getResult() == 0)
+
+    pool.resume()
```

Comment:
I think it would be good to add a test case where more than one task is interrupted. Also, there should be a test case, where only one of two tasks is interrupted, because the interrupt is too late for the first task.